stlab/future.hpp

```diff
@@ -814,23 +814,21 @@ struct when_all_shared {
     std::tuple<boost::optional<Ts>...>  _args;
     future<void>                        _holds[sizeof...(Ts)] {};
     std::atomic_size_t                  _remaining {sizeof...(Ts)};
-    std::atomic_bool                    _error_happend{ false };
-    std::mutex                          _errormutex;
+    std::atomic_bool                    _error_happened{ false };
     boost::optional<std::exception_ptr> _error;
     packaged_task<>                     _f;
 
-    void done() { if (--_remaining == 0 && !_error_happend) _f(); }
+    void done() { if (--_remaining == 0 && !_error_happened) _f(); }
```

Comment:
From my follow on observation - I think the test for _error_happened here is unneeded. _f() gets invoke exactly once - Where all the futures are ready (--_remaining == 0) or when the first future returns an error (in which case _remaining is not decremented).